packages/mobile/src/identity/privacy.ts

```diff
@@ -0,0 +1,138 @@
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { PHONE_NUM_PRIVACY_SERVICE } from 'src/config'
+import { updateE164PhoneNumberSalts } from 'src/identity/actions'
+import { e164NumberToSaltSelector, E164NumberToSaltType } from 'src/identity/reducer'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'identity/privacy'
+const SIGN_MESSAGE_ENDPOINT = '/getSalt' // TODO rename that endpoint
+
+export interface PhoneNumberHashDetails {
+  e164Number: string
+  phoneHash: string
+  salt: string
+}
+
+// Fetch and cache a phone number's salt and hash
+export function* fetchPhoneHashPrivate(e164Number: string) {
+  try {
+    Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Fetching phone hash details')
+    const saltCache: E164NumberToSaltType = yield select(e164NumberToSaltSelector)
+    const cachedSalt = saltCache[e164Number]
+
+    if (cachedSalt) {
+      Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was cached')
+      const phoneHash = getPhoneHash(e164Number, cachedSalt)
+      return { e164Number, phoneHash, salt: cachedSalt }
+    }
+
+    Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was not cached, fetching')
+    const details: PhoneNumberHashDetails = yield call(getPhoneHashPrivate, e164Number)
+    yield put(updateE164PhoneNumberSalts({ [e164Number]: details.salt }))
+    return details
+  } catch (error) {
+    if (error.message === ErrorMessages.SALT_QUOTA_EXCEEDED) {
+      Logger.error(
+        `${TAG}@fetchPrivatePhoneHash`,
+        'Salt quota exceeded, navigating to quota purchase screen'
+      )
+      // TODO nav to quota purchase screen
+    } else {
+      throw new Error(ErrorMessages.SALT_FETCH_FAILURE)
+    }
+  }
+}
+
+// Unlike the getPhoneHash in utils, this leverage the phone number
+// privacy service to compute a secure, unique salt for the phone number
+// and then appends it before hashing.
+async function getPhoneHashPrivate(e164Number: string): Promise<PhoneNumberHashDetails> {
+  const salt = await getPhoneNumberSalt(e164Number)
+  const phoneHash = getPhoneHash(e164Number, salt)
+  return {
+    e164Number,
+    phoneHash,
+    salt,
+  }
+}
+
+async function getPhoneNumberSalt(e164Number: string) {
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Getting phone number salt')
+
+  if (!isE164Number(e164Number)) {
+    throw new Error(ErrorMessages.INVALID_PHONE_NUMBER)
+  }
+
+  // TODO remove when iOS works
+  if (Platform.OS === 'ios') {
+    return 'fakeSalt'
+  }
+
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving blinded message')
+  const base64BlindedMessage = await BlindThresholdBls.blindMessage(e164Number)
+  const base64BlindSig = await postToSignMessage(base64BlindedMessage)
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving unblinded signature')
+  const base64UnblindedSig = await BlindThresholdBls.unblindMessage(base64BlindSig)
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Converting sig to salt')
+  return getSaltFromThresholdSignature(base64UnblindedSig)
+}
```

Comment:
This looks like the core logic in this new system.
I understand the output but I honestly don't understand what it does internally.
Could we explain this in a somewhat approachable way? 😅 
I know this is very new, but would be cool to document it publicly.

Reply:
One of the epics for this milestone is to create public documentation for this :)